src/overlays/actors/ovl_Door_Ana/z_door_ana.c

```diff
@@ -22,19 +33,184 @@ const ActorInit Door_Ana_InitVars =
     (ActorFunc)DoorAna_Update,
     (ActorFunc)DoorAna_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/func_80993EF0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/DoorAna_Init.s")
+// initial collision data
+static ColliderCylinderInit colliderInit =
+{
+    0x0A, 0x00, 0x09, 0x00, 
+    0x00, 0x01, 0x00, 0x00,
+    0x02, 0x00, 0x00, 0x00,
+    0x00000000, 0x00, 0x00, 
+    0x00, 0x00, 0x00000048, 
+    0x00, 0x00, 0x00, 0x00, 
+    0x00, 0x01, 0x00, 0x00, 
+    0x0032, 0x000A, 0x0000, 
+    0x0000, 0x0000, 0x0000
+};
+
+// array of entrance table entries to grotto destinations
+static s16 entrances[] = {
+    0x036D, 0x003F, 0x0598, 0x059C, 
+    0x05A0, 0x05A4, 0x05A8, 0x05AC, 
+    0x05B0, 0x05B4, 0x05B8, 0x05BC, 
+    0x05C0, 0x05C4, 0x05FC
+};
+
+// display list
+extern Gfx* D_05001390;
+
+// sets current actionFunc to be ran on next update call
+static void func_80993EF0(DoorAna* this, ActorFunc func)
+{
+    this->actionFunc = func;
+}
+
+static void DoorAna_Init(DoorAna* this, GlobalContext* globalCtx)
+{
+    ColliderCylinderMain* collider;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Door_Ana/DoorAna_Destroy.s")
+    this->actor.shape.rot.z = 0;
+    this->actor.shape.rot.y = this->actor.shape.rot.z;
+    // init block for grottos that are initially "hidden" (require explosives/hammer/song of storms to open)
+    if ((this->actor.params & 0x300) != 0)
+    {
+        // only allocate collider for grottos that need bombing/hammering open
+        if ((this->actor.params & 0x200) != 0)
+        {
+            collider = &this->collider;
+            ActorCollider_AllocCylinder(globalCtx, collider);
+            ActorCollider_InitCylinder(globalCtx, collider, &this->actor, &colliderInit);
+        }
+        else
+        {
+            this->actor.flags |= 0x10;
+        }
+        Actor_SetScale(&this->actor, 0);
+        func_80993EF0(this, (ActorFunc)&func_80993FEC);
+        this->actor.unk_1F = 0;
+        return;
```

Comment:
This is very likely not an actual return, and the second block is an else of the outer if. Try it out and see if it works